streamz/core.py

```diff
@@ -497,6 +534,28 @@ def to_batch(self, **kwargs):
         from .batch import Batch
         return Batch(stream=self, **kwargs)
 
+    def _retain_refs(self, metadata):
+        if not isinstance(metadata, list):
```

Comment:
Again, no docstrings here or even comments; and more `isinstance` calls than I am comfortable with. `metadata` is a dict or a list of dict?

Reply:
While I agree with this, I'll need to figure out a way to reduce these.

Reply:
I have added docstrings here and simplifed the code by assuming that metadata is always a list